javascript/app-discover/src/state/hooks/useCausalDiscoveryRunner.ts

```diff
@@ -154,70 +156,98 @@ export function useCausalDiscoveryRunner() {
 		[cancelLastDiscoveryResultPromise, setLastDiscoveryResultPromise],
 	)
 
+	const runDiscovery = useCallback(async () => {
+		setErrorMessage(undefined)
+
+		const discoveryPromise = await createDiscoveryPromise(
+			dataset,
+			inModelCausalVariables,
+			causalDiscoveryConstraints,
+			causalDiscoveryAlgorithm,
+			updateProgress,
+			algorithmParams,
+		)
+
+		try {
+			setIsLoading(true)
+			const results = await discoveryPromise.promise!
+
+			// only update if the promise is not canceled
+			if (discoveryPromise.isFinished()) {
+				setCausalDiscoveryResultsState(results)
```

Comment:
`await results`? Is the result a promise or a result object?

Reply:
it is a result object